Woche2/substitution/substitution.c

```diff
@@ -0,0 +1,65 @@
+#include <cs50.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+
+int main(int argc, string argv[]) {
+
+    int count_key_characters = 0;
+    int count_repeated_characters = 0;
+
+    if (argc != 2) {
+        printf("Usage: ./substitution key\n");
+        return 1;
+    }
+
+    if(strlen(argv[1]) != 26) {
+        printf("Key must contain 26 characters.\n");
+        return 1;
+    }
+
+    for (int i = 0, n = strlen(argv[1]); i < n; i++) {
+        bool CurrentLetterIsAlphabetic = isalpha(argv[1][i]);
+        if(CurrentLetterIsAlphabetic) {
+            count_key_characters++;
+        } else if(count_key_characters != strlen(argv[1])) {
+            printf("Key must only contain alphabetic characters.\n");
+            return 1;
+        }
```

Comment:
```suggestion
        if(!CurrentLetterIsAlphabetic) {
            printf("Key must only contain alphabetic characters.\n");
            return 1;
        }
```
This would simplify this check as you want this to end the moment you find a non-alphalbetic character.